### PR TITLE
AGM Location Change 5th PR for styling issues

### PR DESCRIPTION
**Pull request by JazzarKarim**
*Issue #:* bcgov/entity#18161 + bcgov/entity#16764

*Description of changes:*
Fixed what was leftover from previous PR (https://github.com/bcgov/business-filings-ui/pull/556):
  - Fixed validation red error borders
  - Fixed certify button ripple

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the business-filings-ui license (Apache 2.0).

### CODE REVIEW

File: src/components/common/Certify.vue
Comment (JazzarKarim) [inline src/components/common/Certify.vue:179]: Had to configure this manually from scratch.
Before:
Screenshot (112)

After:
Screenshot (111)

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-filings-dev--pr-559-ku35gn15.web.app

**Comment by JazzarKarim** [comment]
For the Validation red border errors;
Before:
before validations

After:
after validation errors

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-filings-dev--pr-559-ku35gn15.web.app

File: src/views/AgmLocationChg.vue
Comment (severinbeauvais) [inline src/views/AgmLocationChg.vue:93]: Screenshot, please.

Comment (severinbeauvais) [reply src/views/AgmLocationChg.vue:93]: Oops, nvm.

**Comment by severinbeauvais** [comment]
> For the Validation red border errors; Before:
> 
> After: 

I'm thinking you should remove the top-left and bottom-right border radius for the red border. That's what our UI/UX guys wanted in previous filings.

File: src/components/common/Certify.vue
Comment (severinbeauvais) [reply src/components/common/Certify.vue:179]: Your change makes this checkbox behave differently from other checkboxes. Why not do something like this?

image

Comment (JazzarKarim) [reply src/components/common/Certify.vue:179]: Yup, I thought the circle thing was wrong. It's just the location of the ripple was wrong. I was thinking of something else.

After latest commit:
Screenshot (113)

**Comment by JazzarKarim** [comment]
> > For the Validation red border errors; Before:
> > After:
> 
> I'm thinking you should remove the top-left and bottom-right border radius for the red border. That's what our UI/UX guys wanted in previous filings.

Great catch Sev! After latest commit:
borders radii

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-filings-dev--pr-559-ku35gn15.web.app

File: src/components/common/Certify.vue
Comment (severinbeauvais) [reply src/components/common/Certify.vue:179]: I'm ok to resolve this conversation.

**Comment by severinbeauvais** [comment]
Please check for console errors when  you first land on the page and click File Now.

**Comment by severinbeauvais** [comment]
> Please check for console errors when you first land on the page and click File Now.

Please test and see what you get.

**Comment by JazzarKarim** [comment]
> > Please check for console errors when you first land on the page and click File Now.
> 
> Please test locally and see what you get.

Is it the "[Vue warn]: Avoid mutating a prop directly since the value will be overwritten whenever the parent component re-renders. Instead, use a data or computed property based on the prop's value. Prop being mutated: "error"" errors?

If I'm not mistaken, that has been the case for a while. I can try and fix all of them in this PR. Let me know Sev.

**Comment by severinbeauvais** [comment]
Do not merge until Business Digital Credentials is out of Dev environment.

**Comment by JazzarKarim** [comment]
If non-staff:
if non staff certify

If staff:
if staff

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-filings-dev--pr-559-ku35gn15.web.app

File: src/views/AgmLocationChg.vue
Comment (severinbeauvais) [inline src/views/AgmLocationChg.vue:461]: Are you sure we didn't already load this data in App.vue?

**Comment by severinbeauvais** [comment]
> On load, If non-staff: (snip)
> 
> If staff: (snip)

In both cases, the name is editable, right?

File: src/views/AgmLocationChg.vue
Comment (JazzarKarim) [reply src/views/AgmLocationChg.vue:461]: We use this same function in App.vue but nothing is being set in the store. I had two choices, either set the logged in user's name in the store or add this here. I went with this route. Let me know if I should go with the first one Sev (or if you think something else is better).

**Comment by JazzarKarim** [comment]
> > On load, If non-staff: (snip)
> > If staff: (snip)
> 
> In both cases, the name is editable, right?

Yes!

File: src/views/AgmLocationChg.vue
Comment (severinbeauvais) [reply src/views/AgmLocationChg.vue:461]: It's better for an API call to be made as few times as possible.

Also, keeping it here could slow down the page's loading by up to 30 seconds (or whatever the network timeout is), and I was hoping we could get rid of the loading spinner and associated plumbing.

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-filings-dev--pr-559-ku35gn15.web.app

File: src/App.vue
Comment (severinbeauvais) [inline src/App.vue:591]: Don't do this here. It's a "side effect" -- not the purpose of this function; no one would look here to see where user full name is set.

Also, I think you should save the entire user info object into store -- we've already done the work of fetching it; there might be other info in there we could use later.

File: src/stores/authenticationStore.ts
Comment (severinbeauvais) [inline src/stores/authenticationStore.ts:26]: Either move this to rootStore.ts or move the property (and setter) to this store.

My vote is to move this to the root store. This store uses `RootStateIF` (unfortunately named) which contains user account and auth data. Meanwhile the root store (which uses `StateIF`) contains more "info" type data.

Comment (JazzarKarim) [reply src/stores/authenticationStore.ts]: Sev, I moved it to the rootStore and did the changes accordingly.

**Comment by JazzarKarim** [comment]
Last commit, I'm doing: https://github.com/bcgov/entity/issues/16764

Non-staff:
new certify consent

Staff:
new certify consent staff

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-filings-dev--pr-559-ku35gn15.web.app

File: src/interfaces/current-user-interface.ts
Comment (severinbeauvais) [inline src/interfaces/current-user-interface.ts:10]: This interface is used by the authentication store, which gets its data from the KC token (not from the auth "users/@me" call).

Sample data:
image

Don't use THIS interface it you want to type the userInfo object.

File: src/views/ConsentContinuationOut.vue
Comment (severinbeauvais) [inline src/views/ConsentContinuationOut.vue:196]: I thought it was supposed to be editable in all cases?

File: src/views/ConsentContinuationOut.vue
Comment (severinbeauvais) [inline src/views/ConsentContinuationOut.vue:512]: If getUserInfo is null, this would show `undefined undefined`.

**Review by severinbeauvais — CHANGES_REQUESTED** [review]
~Sorry, you broke an interface.~ _fixed now_

File: src/views/ConsentContinuationOut.vue
Comment (JazzarKarim) [reply src/views/ConsentContinuationOut.vue:196]: The ticket says for end-user and follow SP/GP registration's way. I checked there and it was only editable if staff.

File: src/App.vue
Comment (severinbeauvais) [reply src/App.vue]: I'm resolving this conversation because the initial issue is fixed.

However, you have created another issue, but it's captured in another conversation so it's still OK to resolve this one.

File: src/views/ConsentContinuationOut.vue
Comment (severinbeauvais) [reply src/views/ConsentContinuationOut.vue:196]: Please check with Mihai. I think there's something underway to allow it to be editable for regular users too.

File: src/views/ConsentContinuationOut.vue
Comment (severinbeauvais) [reply src/views/ConsentContinuationOut.vue:512]: I like this fix 👍

Comment (severinbeauvais) [reply src/views/ConsentContinuationOut.vue:512]: Oh, please fix the other place you do this :)

File: src/interfaces/state-interface.ts
Comment (severinbeauvais) [inline src/interfaces/state-interface.ts:22]: Yeah, I did the same thing. Looks good.

File: src/views/ConsentContinuationOut.vue
Comment (JazzarKarim) [reply src/views/ConsentContinuationOut.vue:196]: I sent Mihai a message Sev. Waiting for his answer.

Comment (JazzarKarim) [reply src/views/ConsentContinuationOut.vue:196]: Sev, Mihai confirmed that it's OK (https://github.com/bcgov/entity/issues/16764#issuecomment-1791493214).

Comment (severinbeauvais) [reply src/views/ConsentContinuationOut.vue:196]: Yup. And it's editable for the AGM filings, right?

**Review by severinbeauvais — APPROVED** [review]
LGTM.

Get yourself some other approvals and let's merge this puppy.

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-filings-dev--pr-559-ku35gn15.web.app

File: src/views/ConsentContinuationOut.vue
Comment (JazzarKarim) [reply src/views/ConsentContinuationOut.vue:196]: For AGM Filings, it's editable in all cases. However, I'm going to double check with Mihai just in case something has changed.

File: tests/unit/AgmLocationChg.spec.ts
Comment (ketaki-deodhar) [inline tests/unit/AgmLocationChg.spec.ts:148]: This looks something new :)

**Review by ketaki-deodhar — APPROVED** [review]
LGTM!

File: tests/unit/AgmLocationChg.spec.ts
Comment (JazzarKarim) [reply tests/unit/AgmLocationChg.spec.ts:148]: This function existed before but we rarely use it! It's found in the tests for PDF Upload for example when we need to wait for DOM updates.

The other way to fix it is have 2 `await Vue.nextTick()`. This function takes place instead of having these duplicate lines.

File: src/views/ConsentContinuationOut.vue
Comment (JazzarKarim) [reply src/views/ConsentContinuationOut.vue:196]: Going to merge now. Mihai said there'll be another ticket in case something changes. I think the requirement for that is still in the works by him. If something needs to change, we'll fix it there in the another ticket or I'll just open another PR.

**Pull request merged**
